invenio_records_rest/serializers/base.py

```diff
@@ -31,22 +31,137 @@
 import pytz
 
 
-class PreprocessorMixin(object):
+class SerializerMixinInterface(object):
+    """Mixin serializing records.
+
+    A record serializer should inherit subclasses of
+    - SerializerMixinInterface
+    - TransformerMixinInterface
+    - PreprocessorMixinInterface
+
+    This class forwards its input to the transformer mixin and serializes its
+    result.
+    """
+
+    def serialize(self, pid, record, links_factory=None, **kwargs):
+        """Serialize a single record and persistent identifier.
+
+        This method should delegate the record transformation to the
+        transformer mixin's transform_record method.
+
+        :param pid: Persistent identifier instance.
+        :param record: Record instance.
+        :param links_factory: Factory function for record links.
+        """
+        raise NotImplementedError()
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None, **kwargs):
+        """Serialize a search result.
+
+        This method should delegate each record search hit transformation
+        to the transformer mixin's transform_search_hit method.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        :param item_links_factory: Factory function for record links.
+        """
+        raise NotImplementedError()
+
+    def serialize_oaipmh(self, pid, record):
+        """Serialize a single record for OAI-PMH.
+
+        :param pid: The :class:`invenio_pidstore.models.PersistentIdentifier`
+            instance.
+        :param record: The :class:`invenio_records.api.Record` instance.
+        :returns: The object serialized.
+        """
+        raise NotImplementedError()
```

Comment:
@lnielsen Here is the `serialize_oaipmh` method. It is not implemented by the JSONSerializer as it does not support OAIPMH

Reply:
I think that's fine. If used it will correctly say that it's not compatible.